packages/sse/src/decorators.ts

```diff
@@ -0,0 +1,12 @@
+import type { TerminalEventDecorator } from "../generated-defs/TypeSpec.SSE.js";
+import { SSEStateKeys } from "./lib.js";
+
+/** @internal */
+export const namespace = "TypeSpec.SSE";
+
+export const $terminalEvent: TerminalEventDecorator = (context, target) => {
+  // TODO: Add a check that the target's parent Union is decorated with `@events`.
```

Comment:
missing check?

Reply:
Ah - I did this check in the `$onValidate` method instead, can remove this TODO.